src/utils.jl

```diff
@@ -17,6 +17,6 @@ ofeltype(x, y) = convert(float(eltype(x)), y)
 
 _eps(T::Type{<:AbstractFloat}, e) = T(e)
 # catch complex and integers
-_eps(T::Type{<:Number}, e) = _eps(real(float(T)), e) 
+_eps(T::Type{<:Number}, e) = real(float(T))(e) 
 # avoid small e being rounded to zero
 _eps(T::Type{Float16}, e) = e == 0 ? T(0) : max(T(1e-7), T(e))
```

Comment:
In which case we can also delete this comment.
```suggestion
_eps(T::Type{Float16}, e) = iszer(e) ? T(0) : max(T(1e-7), T(e))
```

or perhaps clearer
```suggestion
_eps(::Type{Float16}, e) = iszero(e) ? Float16(0) : max(Float16(1e-7), Float16(e))
```